base/sparse/umfpack.jl

```diff
@@ -113,12 +148,13 @@ function lufact{Tv<:UMFVTypes,Ti<:UMFITypes}(S::SparseMatrixCSC{Tv,Ti})
     finalizer(res, umfpack_free_symbolic)
     umfpack_numeric!(res)
 end
-lufact(A::SparseMatrixCSC) = lufact(float(A))
+lufact{T<:Real}(A::SparseMatrixCSC{T}) = lufact(map(Float64, A))
```

Comment:
There doesn't seem to be a big difference for dense arrays but for sparse matrices it doesn't seem like a good advice. However, I think it will soon be time to remove the deprecation warning. As I remember it, `float64` was deprecated in 0.4.0 so in 0.5.0 it should be an error if we follow the normal practice.